### DISCUSSION TITLE
Query Parameters

### DISCUSSION THREAD

**Discussion by jamesmr89** [shlinkio/discussions#86] [Help wanted] [answered]
Hi, 

Brand new to shlink and trying to see if it will meet needs to replace current custom tool being used.

Can we do something like

https://shdomain/slug/12345 and map it to https://mysite/mypage.aspx?param=12345

Thanks,

**Comment by acelaya** [comment, marked as answer]
Hey! Yes, this can be done with Shlink.

Shlink has a feature called "special redirects", which lets you capture requests to not explicitly configured URLs (I assume you want the 12345 part to be dynamic), and then redirect to somewhere else.

These redirects support placeholders, so you would be able to redirect to a dynamic URL containing the original short code as a query param.

You can see more details in the docs: https://shlink.io/documentation/advanced/special-redirects/#placeholders

**Reply by jamesmr89** [reply]
Hmm, not sure this will work or at least maybe I don't fully understand

I want to set the custom slug to something like test/(\d+) and have it redirect to: https://mysite/testresults.aspx?testid=12345

when trying the current implementation when setting placeholders i'm getting redirected to the literal:
https://example.com/?p=%7BORIGINAL_PATH%7D&d=%7BDOMAIN%7D

Is there something I'm missing when configuring?  

Thanks,

**Reply by acelaya** [reply]
What I explained above is not done through a regular short URL. You need to configure the special redirects/not found redirects, which are explained in the page I shared above.
